runtime/development/src/lib.rs

```diff
@@ -1458,12 +1459,99 @@ impl_runtime_apis! {
 		}
 	}
 
-	impl runtime_common::AnchorApi<Block> for Runtime {
+	impl runtime_common::apis::AnchorApi<Block, Hash, BlockNumber> for Runtime {
 		fn get_anchor_by_id(id: Hash) -> Option<AnchorData<Hash, BlockNumber>> {
 			Anchor::get_anchor_by_id(id)
 		}
 	}
 
+	impl runtime_common::apis::LoansApi<Block, PoolId, InstanceId, Balance> for Runtime {
+		fn nav(id: PoolId) -> Option<Balance> {
+			// TODO: Error out if pool does not exists in the loans-pallet
+			if !pallet_pools::Pool::<Runtime>::get(id).is_some() {
+				return None;
+			}
+			pallet_loans::Pallet::<Runtime>::update_nav_of_pool(id)
+				.ok()
+				.map(|(latest, _)| latest.into())
+		}
+
+		fn max_borrow_amount(id: PoolId, loan_id: InstanceId) -> Option<Balance> {
+			let now = <pallet_timestamp::Pallet::<Runtime> as UnixTime>::now().as_secs();
+			pallet_loans::Loan::<Runtime>::get(id, loan_id)
+				.map(|loan_details| loan_details.max_borrow_amount(now).into())
+		}
+	}
+
+	impl runtime_common::apis::PoolsApi<Block, PoolId, TrancheId, Balance, CurrencyId, Rate> for Runtime {
+		fn pool_value(pool_id: PoolId) -> Option<Balance> {
+			let pool = pallet_pools::Pool::<Runtime>::get(pool_id)?;
+			let nav: Balance = pallet_loans::Pallet::<Runtime>::update_nav_of_pool(pool_id)
+				.ok()
+				.map(|(latest, _)| latest.into())?;
+
+			Some(pool.reserve.total.saturating_add(nav))
+		}
+
+		fn pool_currency(pool_id: PoolId) -> Option<CurrencyId>{
```

Comment:
I was thinking we just add RPC methods for anything that we can't easily retrieve from storage, so this could be removed. Or was there a specific reason you also added this?

Reply:
Yes. Was just in the Flow. Did not think about it. 

For the tranche currencies. They are also retrievable from storage, but I think it is more to complex with the ids etc. so I would keep those.

Reply:
@offerijns I thought about it a little. And actually I would like to have more rpcs and keep the pool-currency here. I.e. in my opinion it boils down to, abstracting the storage structure from the front-end. Where ever possible. I am not 100% sure how storage fetches work in polkadot-js, but I guess they have a similar overhead. 

Having rpc-abstractions on top of storage would allow us to have a more stable interface for the UI, even if we change logic.